flytekit/extras/tasks/shell.py

```diff
@@ -210,8 +217,22 @@ def execute(self, **kwargs) -> typing.Any:
             print(gen_script)
             print("\n==============================================\n")
 
+        if self._env is not None:
+            for k, v in self._env.items():
+                os.environ[k] = str(v)
```

Comment:
This has lasting side effects right? 

I googled because I wasn't sure, but subprocesss doesn't support a specific way of avoiding this. So, [recommendation](https://stackoverflow.com/questions/2231227/python-subprocess-popen-with-a-modified-environment) is to use os.environ.copy() and then return the enviornment to normal afterwards (I would write a quick test for this)